### PR TITLE
Bug surface format & AHardwareBuffer

### PR DESCRIPTION
**Pull request by locke-lunarg**

### CODE REVIEW

File: framework/encode/vulkan_state_writer.cpp [hunk 1]
```diff
@@ -1062,10 +1062,16 @@ void VulkanStateWriter::ProcessHardwareBuffer(format::HandleId memory_id,
...
         WriteCreateHardwareBufferCmd(memory_id, hardware_buffer, plane_info);
-        WriteFillMemoryCmd(memory_id, 0, allocation_size, data);
+
+        if (data != nullptr)
+        {
+            WriteFillMemoryCmd(memory_id, 0, allocation_size, data);
+        }
+        else
+        {
+            GFXRECON_LOG_WARNING("AHardwareBuffer_unlock warning: data is nullptr");
```
Comment (bradgrantham-lunarg) [inline framework/encode/vulkan_state_writer.cpp:1073]: Should this warn about `AHardwareBuffer_lock`, since I think that's responsible for setting `data`?

**Review by bradgrantham-lunarg — APPROVED** [review]
Looks great other than my question about the print!

File: framework/encode/vulkan_state_writer.cpp (hunk 1 above)
Comment (bradgrantham-lunarg) [reply framework/encode/vulkan_state_writer.cpp:1073]: That is to say, this LOG_WARNING mentions `..._unlock`, but the function call is `..._lock`.

Comment (locke-lunarg) [reply framework/encode/vulkan_state_writer.cpp:1073]: Honestly, I have no idea what the good log is here. Could you write that for me? Thanks.

**Review by bradgrantham-lunarg — APPROVED** [review]

**Pull request merged**